### PR TITLE
Fixes #11553 - introduce session actors to avoid blocking on opening connections

### PR DESCRIPTION
**Pull request by iNecas**
The commands now work more independently, not having wait while other 
connections are getting opened.

### CODE REVIEW

File: lib/smart_proxy_remote_execution_ssh/command_update.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+module Proxy::RemoteExecution::Ssh
+  # update sent back to the suspended action
+  class CommandUpdate
+    attr_reader :buffer, :exit_status
+
+    def initialize(buffer)
+      @buffer      = buffer
```
Comment (ares) [inline lib/smart_proxy_remote_execution_ssh/command_update.rb]: nit: extra whitespace?

File: lib/smart_proxy_remote_execution_ssh/session.rb [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      self.parent
+    end
+
+    def start_termination(*args)
+      super
+      close
+      finish_termination
+    end
+
+    protected
```
Comment (ares) [inline lib/smart_proxy_remote_execution_ssh/session.rb]: I know it's copied, but shouldn't this be private?

File: lib/smart_proxy_remote_execution_ssh/command_update.rb (hunk 1 above)
Comment (iNecas) [inline lib/smart_proxy_remote_execution_ssh/command_update.rb]: White-space commit pushed (I've also updated some alignations on =>?, as I know you don't like it:)

File: lib/smart_proxy_remote_execution_ssh/session.rb (hunk 2 above)
Comment (iNecas) [inline lib/smart_proxy_remote_execution_ssh/session.rb]: right, updated

**Comment by ares** [comment]
ACK

**Comment by iNecas** [comment]
Squashed, merging

**Pull request merged**